app/controllers/dock_boats_controller.rb

```diff
@@ -0,0 +1,6 @@
+class DockBoatsController < ApplicationController
+  def index
+    dock = Dock.find(params[:dock_id])
```

Comment:
do you still need to access that dock variable in the index view? If so it should be an instance variable